backend/src/ee/services/dynamic-secret/providers/models.ts

```diff
@@ -221,9 +232,29 @@ export const LdapSchema = z.union([
   })
 ]);
 
-export const DynamicSecretTotpSchema = z.object({
-  url: z.string().url().trim().min(1)
-});
+export const DynamicSecretTotpSchema = z.discriminatedUnion("configType", [
+  z.object({
+    configType: z.literal(TotpConfigType.URL),
+    url: z
+      .string()
+      .url()
+      .trim()
+      .min(1)
+      .refine((val) => {
+        const urlObj = new URL(val);
+        const secret = urlObj.searchParams.get("secret");
+
+        return Boolean(secret);
+      }, "OTP URL must contain secret field")
+  }),
+  z.object({
+    configType: z.literal(TotpConfigType.MANUAL),
+    secret: z.string().min(1),
```

Comment:
Not sure if ideal to do here but somewhere use `transform` to `replaceAll(" ", "")` to not require users to manually remove spaces from the key?